src/main/java/no/finn/unleash/util/UnleashConfig.java

```diff
@@ -227,7 +265,10 @@ public UnleashConfig build() {
                     sendMetricsInterval,
                     disableMetrics,
                     contextProvider,
-                    synchronousFetchOnInitialisation);
+                    synchronousFetchOnInitialisation,
+                    Optional.ofNullable(scheduledExecutor).orElseGet(UnleashScheduledExecutorImpl::getInstance),
+                    Optional.ofNullable(unleashSubscriber).orElseGet(NoOpSubscriber::new)
```

Comment:
Is no-op the right default, or should it perhaps be something more substantial like Log4JSubscriber? We could put the full responsibility of handling exceptions on the subscriber so that log-statements like https://github.com/Unleash/unleash-client-java/pull/67/files#diff-7fc449c201ef2331c2e8474f2d9ace6eL51 could be removed...

Reply:
yes, I would prefer that we use subscriber for the logs.

The only thing that worries me then is that worries me then is that you end up losing logs if the user decide to register a custom subscriber and only cares about one of the events (e.g. the metrics). 

Maybe this can to be a separate PR?

Reply:
...or maybe the default implementation in the interface can perform logging on `onError`?

Reply:
and maybe we should provide a `DebugSubscriber` people can use to debug Unleash? I see that can be useful from time to time.

Reply:
I like the idea of letting the default UnleashSubscriber.onError() log errors and will refactor towards that. 

Also, the current Log4JSubscriber is already pretty noisy and could probably be used to debug unleash. But i'll add the possibility to adjust the levels directly on the instance to make it more usable.